src/App.tsx

```diff
@@ -1,14 +1,33 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilteringType } from './types/FilteringType';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosLoading, setTodosLoading] = useState(false);
+  const [filter, setFilter] = useState<FilteringType>(FilteringType.all);
+  const [title, setTitle] = useState('');
+  const [todoCard, setTodoCard] = useState<Todo>();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [iconId, setIconId] = useState(0);
```

Comment:
The name of this variable is a bit misleading - you store the id of the selected todo, not the id of some icon. But anyway, this state is redundant - you may use `todoCard.id` instead of `iconId`